### PR TITLE
System events

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Adds an ability to post system events via `SystemGateway`.
  * Eliminates some system commands previously sent to `CommandLifecycleAggregate` in favor of
     imported events.
  * Adds `EventImported` event and its handling in `EntityHistoryAggregate`.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@dmdashenkov, PTAL.

@armiol, FYI. Your review is welcome as always.

File: core/src/test/java/io/spine/core/EmptyClassTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Test
+    @DisplayName("be serializable")
+    void serialize() {
+        SerializableTester.reserializeAndAssert(EmptyClass.instance());
+    }
+
+    @Test
+    @DisplayName("return the same instance")
+    void sameInstance() {
```
Comment (dmdashenkov) [inline core/src/test/java/io/spine/core/EmptyClassTest.java]: Let's extract this into a `SingletonTest`(-`er`) in `testutil`.

File: server/src/main/java/io/spine/server/commandbus/SeveralCommands.java [hunk 2]
```diff
@@ -69,12 +77,11 @@ public SeveralCommands addAll(Iterable<? extends Message> commandMessage) {
     addPosted(MarkCausedCommands.Builder builder, Command command, SystemGateway gateway) {
         CommandId commandId = command.getId();
         builder.addProduced(commandId);
-        markReacted(gateway, commandId);
     }
 
     /**
      * {@inheritDoc}
-     *
+     * &nbsp;
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/commandbus/SeveralCommands.java]: I think, this is too much effort for keeping the warning off the Javadoc.

File: server/src/main/java/io/spine/system/server/DefaultSystemGateway.java [hunk 3]
```diff
@@ -44,9 +46,9 @@
 final class DefaultSystemGateway implements SystemGateway {
 
     /**
-     * The which posts the system events.
+     * The ID of the user used for generating system commands and events.
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/system/server/DefaultSystemGateway.java]: Please add `which is` between `user` and `used`.

File: server/src/main/java/io/spine/system/server/SystemCommandFactory.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+     *        {@code false} otherwise
+     */
+    static CommandFactory newInstance(boolean multitenant) {
+        ActorRequestFactory requestFactory = requestFactory(multitenant);
+        return requestFactory.command();
+    }
+
+    static ActorRequestFactory requestFactory(boolean multitenant) {
+        return multitenant
+        ? newForCurrentTenant()
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/system/server/SystemCommandFactory.java]: Please align the operator.

File: server/src/test/java/io/spine/system/server/ModelInfo.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+package io.spine.system.server;
+
+import io.spine.server.aggregate.model.AggregateClass;
+
+/**
+ * Provides information on model classes for entities of {@link io.spine.system.server} package.
+ *
+ * @author Alexander Yevsyukov
+ */
+public class ModelInfo {
```
Comment (dmdashenkov) [inline server/src/test/java/io/spine/system/server/ModelInfo.java]: Looks `final` and `@Internal`.

Comment (alexander-yevsyukov) [reply server/src/test/java/io/spine/system/server/ModelInfo.java]: It's under `test`, so no need in `@Internal`. But definitely `final`.

File: core/src/test/java/io/spine/core/EmptyClassTest.java (hunk 1 above)
Comment (alexander-yevsyukov) [reply core/src/test/java/io/spine/core/EmptyClassTest.java]: Such a utility should be a part of `base`. Here's the extension request (https://github.com/SpineEventEngine/base/issues/186). It also contains the request for changing `EmptyClassTest` once the test utils are ready.

File: client/src/test/java/io/spine/client/CommandFactoryTest.java [hunk 6]
```diff
@@ -122,6 +96,7 @@ void withOwnTenantId() {
     @DisplayName("throw ValidationException when creating command")
     class NotAccept {
 
+        @SuppressWarnings("unused")
         private final RequiredFieldCommand invalidCommand =
```
Comment (armiol) [inline client/src/test/java/io/spine/client/CommandFactoryTest.java]: In fact, it _may_ used in the methods below.

Please see those. They both have the same _local_ variable defined. So maybe we could kill those local variables instead. And remove this suppression.

File: server/src/main/java/io/spine/server/security/InvocationGuard.java [hunk 7]
```diff
@@ -37,18 +37,39 @@ private InvocationGuard() {
...
         if (!allowedCallers.contains(callingClass.getName())) {
-            String msg = format(
-                    "The class %s is not allowed to perform this operation.", callingClass
-            );
-            throw new SecurityException(msg);
+            throw notAllowedCaller(callingClass);
         }
     }
+
+    private static SecurityException notAllowedCaller(Class callingClass) {
```
Comment (armiol) [inline server/src/main/java/io/spine/server/security/InvocationGuard.java]: `nonAllowedCaller`.

**Pull request merged**